src/plugin.js

```diff
@@ -1,23 +1,84 @@
 import videojs from 'video.js';
+import QualityLevel from './quality-level.js';
+import QualityLevelList from './quality-level-list.js';
 
-// Default options for the plugin.
-const defaults = {};
+const noop = () => {};
 
-/**
- * Function to invoke when the player is ready.
- *
- * This is a great place for your plugin to initialize itself. When this
- * function is called, the player will have its DOM and child components
- * in place.
- *
- * @function onPlayerReady
- * @param    {Player} player
- * @param    {Object} [options={}]
- */
-const onPlayerReady = (player, options) => {
-  player.addClass('vjs-contrib-quality-levels');
+const getHlsPlaylistIndex = function(media, playlists) {
+  for (let i = 0; i < playlists.length; i++) {
+    let playlist = playlists[i];
+
+    if (playlist.resolvedUri === media.resolvedUri) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+const setupHlsHandlers = function(qualityLevels, hls) {
+  hls.playlists.on('loadedmetadata', () => {
+    let selectedPlaylist = hls.playlists.media();
+    let selectedIndex = getHlsPlaylistIndex(selectedPlaylist,
+      hls.playlists.master.playlists);
+
+    hls.representations().forEach((rep) => {
+      qualityLevels.addQualityLevel(new QualityLevel(rep));
+    });
+
+    qualityLevels.selectedIndex_ = selectedIndex;
+    qualityLevels.trigger({
+      selectedIndex,
```

Comment:
I thought it would be useful to pass along the new info with the event, but if we want to stick to the style of just triggering the event without adding a property, we can do that.